src/compositor/output/output.rs

```diff
@@ -40,5 +44,34 @@ impl OutputHandler for Output {
                 }
             }).expect("Could not render views")
         }
+        error!("drawins: {:?}", drawins.len());
+        for &(ref drawin, geometry) in drawins.iter() {
+            // NOTE No need to check if it's visible since if it's in the list
+            // then it's implicitly visible to the user
+            let mut lock = drawin.image();
+            let mut image = match lock {
+                Err(_) => continue,
+                Ok(ref none) if none.is_none() => continue,
+                Ok(ref mut image) => image.as_mut().unwrap()
+            };
+            {
+                let cr = Context::new(&*image);
+                cr.set_source_surface(&*image, 0.0, 0.0);
+                cr.paint();
```

Comment:
Uhm... is this painting a surface to itself?
- Why? (Copy&paste from something older?)
- That does not have defined semantics in Cairo.

Reply:
Yeah no idea why that code is there...Must be _oolllllddd_ code from the old wlc rendering code.